cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -673,6 +673,23 @@ func getEnvWithDefaultInt64(k string, defaultVal int64) int64 {
 	return i
 }
 
+func initializeCompoundShardCounter(indexDir string) {
+	fns, err := filepath.Glob(filepath.Join(indexDir, "*.zoekt"))
```

Comment:
I was reading the code and I don't think there's a big difference between io/fs.Glob and path/filepath.Glob. The former was introduced more recently, so it's the `io.FS` aware version of Glob, with the extra argument.

I think you can do the filtering of the compound- prefix here already.

```suggestion
	fns, err := filepath.Glob(filepath.Join(indexDir, "compound-*.zoekt"))
```

Reply:
And then there's no need to loop over the fns: `metricNumberCompoundShards.Set(float64(len(fns)))